### ISSUE TITLE
Question for `heat_conduction_1d_uniform_bar` example

### ISSUE THREAD

**Issue by Saransh-cpp** [lululxvi/deepxde#399]
Hi, while training the model for `heat_conduction_1d_uniform_bar` example, I get the following output - 
```
Compiling model...
'compile' took 0.001081 s

Training model...

WARNING:tensorflow:AutoGraph could not transform <function main.<locals>.<lambda> at 0x000002AF673BFAF0> and will run it as-is.
Cause: could not parse the source code of <function main.<locals>.<lambda> at 0x000002AF673BFAF0>: no matching AST found
To silence this warning, decorate the function with @tf.autograph.experimental.do_not_convert
WARNING: AutoGraph could not transform <function main.<locals>.<lambda> at 0x000002AF673BFAF0> and will run it as-is.
Cause: could not parse the source code of <function main.<locals>.<lambda> at 0x000002AF673BFAF0>: no matching AST found
To silence this warning, decorate the function with @tf.autograph.experimental.do_not_convert
WARNING:tensorflow:AutoGraph could not transform <function main.<locals>.<lambda> at 0x000002AF673BFC10> and will run it as-is.
Cause: could not parse the source code of <function main.<locals>.<lambda> at 0x000002AF673BFC10>: no matching AST found
To silence this warning, decorate the function with @tf.autograph.experimental.do_not_convert
WARNING: AutoGraph could not transform <function main.<locals>.<lambda> at 0x000002AF673BFC10> and will run it as-is.
Cause: could not parse the source code of <function main.<locals>.<lambda> at 0x000002AF673BFC10>: no matching AST found
To silence this warning, decorate the function with @tf.autograph.experimental.do_not_convert
Step      Train loss                        Test loss                         Test metric
0         [4.00e-02, 1.01e-01, 3.09e-01]    [4.00e-02, 1.01e-01, 3.09e-01]    []  
1000      [1.23e-03, 3.15e-04, 3.56e-04]    [1.23e-03, 3.15e-04, 3.56e-04]    []  
2000      [7.98e-04, 1.81e-04, 6.04e-05]    [7.98e-04, 1.81e-04, 6.04e-05]    []  
3000      [2.14e-04, 2.86e-05, 2.41e-05]    [2.14e-04, 2.86e-05, 2.41e-05]    []  
4000      [1.59e-04, 2.02e-05, 1.57e-05]    [1.59e-04, 2.02e-05, 1.57e-05]    []  
5000      [1.22e-04, 1.46e-05, 1.14e-05]    [1.22e-04, 1.46e-05, 1.14e-05]    []  
6000      [9.74e-05, 1.24e-05, 9.22e-06]    [9.74e-05, 1.24e-05, 9.22e-06]    []  
7000      [7.93e-05, 1.05e-05, 7.58e-06]    [7.93e-05, 1.05e-05, 7.58e-06]    []  
8000      [6.56e-05, 9.33e-06, 6.63e-06]    [6.56e-05, 9.33e-06, 6.63e-06]    []  
9000      [1.66e-04, 1.35e-04, 9.30e-05]    [1.66e-04, 1.35e-04, 9.30e-05]    []  
10000     [4.66e-05, 7.22e-06, 4.88e-06]    [4.66e-05, 7.22e-06, 4.88e-06]    []  
11000     [4.04e-05, 6.57e-06, 4.35e-06]    [4.04e-05, 6.57e-06, 4.35e-06]    []  
12000     [3.30e-04, 3.76e-04, 2.68e-04]    [3.30e-04, 3.76e-04, 2.68e-04]    []  
13000     [3.19e-05, 5.86e-06, 3.66e-06]    [3.19e-05, 5.86e-06, 3.66e-06]    []  
14000     [2.85e-05, 5.18e-06, 3.33e-06]    [2.85e-05, 5.18e-06, 3.33e-06]    []  
15000     [2.61e-05, 4.85e-06, 3.13e-06]    [2.61e-05, 4.85e-06, 3.13e-06]    []  
16000     [2.40e-05, 4.54e-06, 2.93e-06]    [2.40e-05, 4.54e-06, 2.93e-06]    []  
17000     [2.24e-05, 4.35e-06, 2.85e-06]    [2.24e-05, 4.35e-06, 2.85e-06]    []  
18000     [3.32e-05, 2.64e-05, 2.00e-05]    [3.32e-05, 2.64e-05, 2.00e-05]    []  
19000     [1.94e-05, 3.73e-06, 2.59e-06]    [1.94e-05, 3.73e-06, 2.59e-06]    []  
20000     [1.96e-05, 1.02e-05, 8.69e-06]    [1.96e-05, 1.02e-05, 8.69e-06]    []  

Best model at step 19000:
  train loss: 2.57e-05
  test loss: 2.57e-05
  test metric: []

'train' took 155.507816 s

Compiling model...
'compile' took 0.044615 s

Training model...

Step      Train loss                        Test loss                         Test metric
20000     [1.96e-05, 1.02e-05, 8.69e-06]    [1.96e-05, 1.02e-05, 8.69e-06]    []  
20450     [9.12e-07, 9.59e-08, 7.55e-09]    [9.12e-07, 9.59e-08, 7.55e-09]    []  

Best model at step 20450:
  train loss: 1.02e-06
  test loss: 1.02e-06
  test metric: []

'train' took 108.312722 s
```

My 2 concerns are the warnings and the train and test loss. 

Can someone let me know if the things listed in the warnings are somehow messing up with script or is everything normal?

And also, my test and train loss is identical and the plot overlaps completely, shouldn't the loss have some difference (even a minute one)? Could someone also help me with this?

Edit: I also had some questions about the script written - 
```py
    model.compile("adam", lr=1e-3)
    model.train(epochs=20000)
    model.compile("L-BFGS")
    losshistory, train_state = model.train()
```
Why are we compiling and training the model twice? that too with different optimizers

Thanks!

**Comment by Saransh-cpp** [comment]
Thank you for your response @lululxvi, just a minor doubt, I ran the notebook without making any changes and I still get exact same train and test loss. Is the notebook outdated? As the output in the notebook on GitHub seems correct but when I run it locally or on Colab, the losses are same.

**Comment by lululxvi** [comment]
Yes, the notebook is outdated.

**Comment by Saransh-cpp** [comment]
Thank you so much for all the help @lululxvi, I'll try to fix/update a couple of examples through PRs as I go through them. Closing this issue for now!
